Modules/itertoolsmodule.c

```diff
@@ -301,6 +302,11 @@ pairwise_new_impl(PyTypeObject *type, PyObject *iterable)
     }
     po->it = it;
     po->old = NULL;
+    po->result = PyTuple_Pack(2, Py_None, Py_None);
```

Comment:
`PyTuple_Pack` is slow. I created an issue https://github.com/python/cpython/issues/118222 to discuss this.

An alternative that might be faster:
```suggestion
    PyObject tmp[2] = {Py_None, Py_None};
    po->result = _PyTuple_FromArraySteal(tmp, 2),;
```
(note: untested!)

Reply:
I tested this locally. Since this line is once called once for each call to `pairwise` the performance difference is insignificant. It does matter for the other call to `PyTuple_Pack` (the one in `pairwise_next`), but that is best handled in a different PR.